cli/internal/cmd/spinner.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	spinnerStates = []string{"⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"}
+	dotsStates    = []string{".", "..", "..."}
+)
+
+type Spinner struct {
+	out      *cobra.Command
+	text     string
+	showDots bool
+	delay    time.Duration
+	wg       *sync.WaitGroup
+	stop     int32
+}
+
+func NewSpinner(c *cobra.Command, text string, showDots bool) *Spinner {
+	return &Spinner{
+		out:      c,
+		text:     text,
+		showDots: showDots,
+		wg:       &sync.WaitGroup{},
+		delay:    100 * time.Millisecond,
+		stop:     0,
+	}
+}
+
+func (s *Spinner) Start() {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+	out:
+		for {
+			for i := 0; i < len(spinnerStates); i++ {
```

Comment:
```suggestion
		for i := 0; ; i = (i + 1) % len(spinnerStates) {
```
I would suggest to simplify this loop a little to reduce complexity.